archive_converter.py

```diff
@@ -48,32 +54,21 @@ def _add_messages_to_index(self, new_messages : List[Message]):
 
         # First iterates through all messages to distinguish between 1.) replies and 2.) the start of a thread.
         for message in new_messages:
-            if message.id not in self._messages_seen:
-                self._messages_seen[message.id] = message
+            if not self._message_dao.get(message.id):
+                self._message_dao.store(message)
 
             # If message is a reply, it needs to be associated with an existing thread
             if message.in_reply_to is not None:
                 need_parent.append(message)
                 continue
 
-            # If message doesn't have the LKML style [PATCH v4 00/11],
-            # we don't need to put it in the index.
-            key = message.get_key()
-            if key:
-                self._key_to_thread_start[key] = message
-
         for message in need_parent:
-            if message.in_reply_to not in self._messages_seen:
+            if not self._message_dao.get(message.in_reply_to):
```

Comment:
Since `get()` will eventually query the db, should avoid doing `get()` twice here?

I.e.

```python
parent = self._message_dao.get(message.in_reply_to)
if not parent:
  ...
parent.children.append(message)
```